src/hidimstat/knockoffs.py

```diff
@@ -225,7 +222,7 @@ def model_x_knockoff(
 
     results = parallel(
         delayed(memory.cache(_stat_coefficient_diff))(
-            X, X_tildes[i], y, estimator, fdr, preconfigure_estimator
+            X, X_tildes[i], y, clone(estimator), fdr, preconfigure_estimator
```

Comment:
You're cloning the estimator to avoid side effects ?

Reply:
Yes, I want to avoid some side effects due to the parallelisation. It's better to have independent objects.